nettop: add page (#6254)
tldr-pages/tldr: pages/osx/nettop.md

# nettop

> Display dynamic real-time information about the network.

> 

Reviewer: ```suggestion
More information <>.

```

Leave this open and we will decide on a link

Reviewer 2: We tried to introduce 
> More information: N/A.

for pages which didn't have a link a few months ago. But it failed, since tldl-lint wanted to have `< ... >` around the link. Well, then let's introduce:

```suggestion
More information: <N/A>.
```

Reviewer: No, I am not suggesting no link. I am just leaving it here to be discussed and the gap to be filled.

https://www.unix.com/man-page/osx/1/nettop/ 
https://apps.apple.com/us/app/nettop/id1212061421?mt=12

Are these any good?

Reviewer 2: Actually the second one looks pretty good. It doesn't give you information on the link, but after you downloaded it you have access to the man page

Author: This `nettop` is an OSX builtin command, and preinstalled in system. OSX  user can use  `man nettop` directly.

I think the second link is another different application, but have the same name `nettop` 😂

Reviewer: We should just use the unix link

Author: Compare to `man nettop`, the unix link seems to be outdated or incomplete , this is the result of `man nettop`  in osx: https://pastebin.com/vqJc1et3

Reviewer 2: We already have the `--help` example to get more information, so I think we could just use `<N/A>`. An outdated man page is worse than none at all.

Author: Yes, I agree, An outdated man page is a trap :)

Reviewer 2: Then, could you please apply my suggestion above? I'd be interested if `tldr-lint` complains about it.

Author: OK, I have applied your suggestion just now. Leave the next work to `tldr-lint`.

> - Start nettop monitoring TCP and UDP sockets from all interfaces:

Reviewer 2: ```suggestion
- Monitor TCP and UDP sockets from all interfaces:
```
This would make the descriptions a little shorter and less monotonous. That should  be applied everywhere.

Author: OK, I will change it

`nettop`

- Start nettop monitoring TCP sockets from Loopback interfaces:

> `nettop -m tcp -t loopback`

Reviewer 2: ```suggestion
`nettop -m {{tcp}} -t {{loopback}}`
```
We use the token syntax not only for placeholders, but for anything which can be changed

Author: Got it, thanks 😄

> - Start nettop monitoring specific process:

Reviewer 2: ```suggestion
- Start nettop monitoring a specific process:
```

Reviewer: Does it focus on 1 process, if not then

```suggestion
- Start nettop, monitoring specific processes:
```
would be acceptable

Author: Yes, it focus on 1 process, I should add `a` :)

`nettop -p {{process_id|process_name}}`

> - Display a per-process summary:

Author: Does `display` sounds like only show once and won't refresh?

This command will monitor processes,  display a per-process summary and refresh result periodically.

Reviewer: No, print does though.

Author: OK, I understand, thanks :)

> - Display summary only for per-process:

Reviewer: ```suggestion
- Display a summary only for per-process:
```

Reviewer 2: Another suggestion:
```suggestion
- Display a per-process summary only:
```

Reviewer: That sounds weird

Author: I think `Display a summary for per-process` is good 😆

Maybe we don't need word _only_, I copy the word _only_ from the man page:

```plaintext
     -P       Display per-process summary only, skipping details of open con-
	      nections.  This is equivalent to selecting "Collapse All" in the
	      interactive menu.
```

Reviewer 2: I think `per-process` is used as an adjective here, so `a per-process summary` would be the correct way. Maybe @sbrl could help out here.

`nettop -P`

> - Monitor in logging mode and display 10 samples:

Reviewer 2: From how I understood it in the man page, this command doesn't run periodically right? To avoid using `logging mode`, how about
```suggestion
- Print 10 samples of real-time network information: 
```

Author: Yes, this command run only one time, print samples periodically, default Interval time is 1 second. I don't know if this can be called `real-time` 😂

Reviewer 2: I jut took this part from the command description:

> Display dynamic real-time information about the network.

But we can of course just remove the `real-time`.

Author: Oh, it's my fault... I copy the words `dynamic` and `real-time` from `top` command to description, my apologies.

Maybe we should use the command description from `man nettop`: 
> Display updated information about the network

Author: So, maybe `Print 10 samples of network information` ?

Author: OK, I have applied `Print 10 samples of network information`, 
and change command description to `Display updated information about the network`

> `nettop -P`

Reviewer 2: Who thought it might be a good idea to use both `-p` and `-P` as arguments? 😆

Author: Without these two options, this command will show the detail for all processes, a long page will be displayed on screen.
With `-P` option, user can roughly monitor all processes status, while with `-p` option, user can check all connection detail belong to one process, so I think these two options might to be use separately 😄

Reviewer 2: No, I was talking about that they are very easy to confuse, since they look very similar. 😄

Author: Oh, I misunderstood before. Yes, they are really easy to confuse without man page 😅

`nettop -l {{10}}`

> - Monitor in delta mode, updating every 5 seconds:

Reviewer: ```suggestion
- Monitor in delta mode and update every 5 seconds:
```
Follow the standard above you used

Reviewer 2: I wouldn't say that's necessary. A bit diversity in descriptions makes the more interesting.

Reviewer: No its not, but it makes the page a little more uniformed

> - Monitor in delta mode, updating every 5 seconds:

Reviewer 2: And instead of `delta mode` we could use:
```suggestion
- Monitor changes in the network, updating every 5 seconds:
```

Author: Sorry for leaving for a while because of other work...

How about `Monitor relative changes between every 5 seconds` ?
`delta` mode will show relative change, and default mode will show cumulative change.

I feel `delta mode` is a little bit hard to represent in a short words 😂

Reviewer: The descriptions can be a little longer, as long as it has a purpose like displaying relevant information, keeping it short is nicer for the user but if it needs to be longer, it should be longer.

I wouldn't instantly correlate what delta mode is, but after thinking for around a minute, I remember it means change.

maybe `Monitor differences/changes between every 5 seconds`

Author: OK, so the description could be longer to make the presentation clearer. 

I think `Monitor changes between every 5 seconds` is good.

Reviewer: I don't like `between`

Author: How about `during`?

Reviewer 2: I think that's a case for @sbrl too.

Reviewer: Just remove the between, it still makes complete sense

Author: Hello, @marchersimon, could you help out here, Do you think `Monitor changes every 5 seconds` can be applied?

> - Start nettop in delta mode, updating every 5 seconds:

Reviewer 2: What are `logging` and `delta` mode exactly?

Author: `delta` mode will show delta value between every update rather than show cumulative value.

`loggin` mode will print the raw form periodically rather than redisplay.

I think man page in https://www.manpagez.com/man/1/nettop/ is outdated so it dones't have these options.
I pasted the result of `man nettop` to here: https://pastebin.com/vqJc1et3, hope you cloud check it if it is convenient :)

Reviewer 2: I'm not sure if that's the case for OSX, but if you can download that package it has to exist somewhere online to browse through. These files probably contain also the manpage. Maybe that's worth a try.

Author: It's an OSX builtin command. Sorry, according to this answer: https://stackoverflow.com/a/8670402, this command use private code, maybe we can't find the original online man page, just like `launchctl` and `hdiutil`.

Reviewer 2: https://github.com/tldr-pages/tldr/pull/6254/files#r673909261

Author: OK, I see it 😄

`nettop -d -s {{5}}`

> - List interactive commands while running nettop:

Reviewer: ```suggestion
- While running nettop, list interactive commands:
```

Author: OK, it's clearer than before, I will change it.

Author: Oh, I just found out that GitHub can commit suggestions directly, it's so cool. Looks like I'm still lacking in knowledge about GitHub.

> - See interactive commands while running nettop:

Reviewer 2: ```suggestion
- List interactive commands while running nettop:
```
Not obligatory

Author: Probably the semantics of list is clearer, I will do as you say.

Reviewer 2: You could also use `Display` or `Show` but `See` sounds a bit weird in my opinion.

`h`

> - Display help message:

Reviewer 2: ```suggestion
- Display help:
```
or
```suggestion
- Display a help message:
```

Reviewer: we opt for the top one normally.

Author: OK, I will choose the top one.

# nettop

> Display updated information about the network.

> > More information: <N/A>.

Reviewer: ```suggestion
> More information: <N/A>.
```
